test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -80,96 +73,129 @@ call_log_manager::get_session(const std::string session_id)
 }
 
 void
-call_log_manager::process_call_log_entry(json call_log_entry)
+call_log_manager::call_log_begin_transaction(const json &call_log_entry)
+{
+    const std::string session_id = call_log_entry["session_id"].get<std::string>();
+    session_simulator *session = get_session(session_id);
+    session->begin_transaction();
+}
+
+void
+call_log_manager::call_log_close_session(const json &call_log_entry)
+{
+    const std::string session_id = call_log_entry["session_id"].get<std::string>();
+    session_simulator *session = get_session(session_id);
+
+    /* Remove the session from the connection and the session map. */
+    if (_conn->close_session(session))
+        _session_map.erase(session_id);
+    else
+        std::cerr << "Could not close the session (session ID: " << session_id << ")" << std::endl;
+}
+
+void
+call_log_manager::call_log_open_session(const json &call_log_entry)
+{
+    const std::string session_id = call_log_entry["session_id"].get<std::string>();
+
+    /* We should not open sessions with an ID that is already in use. */
+    if (_session_map.find(session_id) != _session_map.end()) {
+        std::cerr << "Could not open duplicate session, session already exists (session ID: "
```

Comment:
I think we could remove `duplicate` from the error message.

Reply:
This code has been changed in new commits.